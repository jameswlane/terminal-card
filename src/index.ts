import * as boxen from 'boxen';
import chalk from 'chalk';

// Define options for Boxen
const options = {
  borderStyle: 'round',
  margin: 1,
  padding: 1,
};

// Text + chalk definitions
const data = {
  github: chalk.cyan('https://github.com/jameswlane'),
  handle: chalk.cyan('jameswlane'),
  labelCard: chalk.white.bold('      Card:'),
  labelGitHub: chalk.white.bold('    GitHub:'),
  labelLinkedIn: chalk.white.bold('  LinkedIn:'),
  labelTwitter: chalk.white.bold('   Twitter:'),
  labelWeb: chalk.white.bold('       Web:'),
  labelWork: chalk.white.bold('      Work:'),
  linkedin: chalk.cyan('https://www.linkedin.com/in/jameswlane/'),
  name: chalk.white('James W. Lane /'),
  npx: chalk.white('npx jameswlane'),
  twitter: chalk.cyan('https://twitter.com/jameswlane'),
  web: chalk.cyan('https://jameswlane.com'),
  work: chalk.white('Agile Coach at John Deere'),
};

// Actual strings we're going to output
const newline = '\n';
const heading = `${data.name} ${data.handle}`;
const working = `${data.labelWork}  ${data.work}`;
const twittering = `${data.labelTwitter}  ${data.twitter}`;
const githubing = `${data.labelGitHub}  ${data.github}`;
const linkedining = `${data.labelLinkedIn}  ${data.linkedin}`;
const webing = `${data.labelWeb}  ${data.web}`;
const carding = `${data.labelCard}  ${data.npx}`;

// Put all our output together into a single variable so we can use boxen effectively
const output = heading + newline + newline + working + newline + twittering + newline
  + githubing + newline + linkedining + newline + webing + newline + newline + carding;

export default chalk.green(boxen(output, options));
