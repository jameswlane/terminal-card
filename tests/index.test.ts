import src from '../src';

test('Output should match snapshot', () => {
  expect(src).toMatchSnapshot();
});
