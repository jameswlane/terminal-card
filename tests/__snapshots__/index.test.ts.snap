// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output should match snapshot 1`] = `
"[32m[39m
[32m   ┌──────────────────────────────────────────────────────────┐[39m
[32m   │                                                          │[39m
[32m   │   [37mJames W. Lane /[32m [36mjameswlane[32m                             │[39m
[32m   │                                                          │[39m
[32m   │   [37m[1m      Work:[22m[32m  [37mAgile Coach at John Deere[32m                 │[39m
[32m   │   [37m[1m   Twitter:[22m[32m  [36mhttps://twitter.com/jameswlane[32m            │[39m
[32m   │   [37m[1m    GitHub:[22m[32m  [36mhttps://github.com/jameswlane[32m             │[39m
[32m   │   [37m[1m  LinkedIn:[22m[32m  [36mhttps://www.linkedin.com/in/jameswlane/[32m   │[39m
[32m   │   [37m[1m       Web:[22m[32m  [36mhttps://jameswlane.com[32m                    │[39m
[32m   │                                                          │[39m
[32m   │   [37m[1m      Card:[22m[32m  [37mnpx jameswlane[32m                            │[39m
[32m   │                                                          │[39m
[32m   └──────────────────────────────────────────────────────────┘[39m
[32m[39m"
`;
